### PR TITLE
update client-c & make corresponding changes

### PR DESCRIPTION
**Pull request by gengliqi**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/7839

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/StorageDisaggregatedRemote.cpp [hunk 1]
```diff
@@ -232,29 +232,33 @@ void StorageDisaggregated::buildReadTaskForWriteNode(
...
-    const auto resp = call.getResp();
-
-    const DM::DisaggTaskId snapshot_id(resp->snapshot_id());
+    pingcap::kv::RpcCall<pingcap::kv::RPC_NAME(EstablishDisaggTask)> rpc(cluster->rpc_client, req->address());
+    disaggregated::EstablishDisaggTaskResponse resp;
+    grpc::ClientContext client_context;
+    rpc.setClientContext(client_context, DEFAULT_DISAGG_TASK_BUILD_TIMEOUT_SEC);
+    auto status = rpc.call(&client_context, *req, &resp);
+    if (!status.ok())
+        return;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/StorageDisaggregatedRemote.cpp:241]: It should throw exception

Comment (gengliqi) [reply dbms/src/Storages/StorageDisaggregatedRemote.cpp:241]: Addressed

**Comment by JaySon-Huang** [comment]
Can you update client-c to https://github.com/tikv/client-c/commit/2efb4b449081f76def7496d1b9b795e0255b6f82?

File: dbms/src/Storages/S3/S3Common.cpp [hunk 2]
```diff
@@ -243,22 +238,28 @@ disaggregated::GetDisaggConfigResponse getDisaggConfigFromDisaggWriteNodes(
...
-            auto rpc_call = pingcap::kv::RpcCall<disaggregated::GetDisaggConfigRequest>(get_config_req);
             try
             {
-                kv_cluster->rpc_client->sendRequest(send_address, rpc_call, /*timeout=*/2);
-                const auto resp = rpc_call.getResp();
-                RUNTIME_CHECK(resp->has_s3_config(), resp->ShortDebugString());
+                pingcap::kv::RpcCall<pingcap::kv::RPC_NAME(GetDisaggConfig)> rpc(kv_cluster->rpc_client, send_address);
+
+                grpc::ClientContext client_context;
+                rpc.setClientContext(client_context, 2);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/S3/S3Common.cpp:246]:
```suggestion
                rpc.setClientContext(client_context, /*timeout=*/2);
```

**Review by JaySon-Huang — COMMENTED** [review]
Mostly LGTM

**Comment by gengliqi** [comment]
> Can you update client-c to [tikv/client-c@2efb4b4](https://github.com/tikv/client-c/commit/2efb4b449081f76def7496d1b9b795e0255b6f82)?

Updated.

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

**Pull request merged**